pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -277,7 +282,7 @@ func setKubeAPIServerArg(arg map[string]string, cluster *v1.Cluster) {
 
 // MachineSelectorFileForPSA generates an RKEProvisioningFiles that mounts the secret which contains
 // the generated admission configuration file to the control plane node
-func MachineSelectorFileForPSA(secretName, mountPath string) rkev1.RKEProvisioningFiles {
+func MachineSelectorFileForPSA(secretName, mountPath, hash string) rkev1.RKEProvisioningFiles {
```

Comment:
nit: You don't need to change anything, but could we have used pointers for all our calls using `rkev1.RKEProvisioningFiles` purely asking out of curiosity definitely not a big deal since the struct only has two pointers in it.

Reply:
The only reason for returning an instance is because the Cluster contains a list of `rkev1.RKEProvisioningFiles`, not pointer to `rkev1.RKEProvisioningFiles`, so it makes some operations easy. 

However, as the implementation iterates, I feel it makes more sense to return a pointer.  

The code is updated now.